File: scripts/ros_erle_statusled_client.py
#!/usr/bin/python
import rospy
import sys
from ros_erle_statusled.msg import *

red = 0
blue = 0

def talker():
    pub = rospy.Publisher("statusled", statusled, queue_size=10)
    rospy.init_node("statusled", anonymous=True)
    msg = statusled()
    msg.blue = blue
    msg.red = red
    msg.green = True
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        pub.publish(msg)
        rate.sleep()

def usage():
    return "%s [red blue]"%sys.argv[0]

if __name__ == "__main__":

    if len(sys.argv) == 3:
        red = int(sys.argv[1])
        blue = int(sys.argv[2])
    else:
        print usage()
        sys.exit(1)
    talker();

File: src/ros_erle_statusled_client.cpp
#include "ros/ros.h"
#include "ros_erle_statusled/statusled.h"
#include <cstdlib>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "ros_erle_statusled_client");
  if (argc != 3){
    ROS_INFO("usage: ros_erle_statusled_client X Y");
    return 1;
  }

  ros::NodeHandle n;
  ros::Publisher led_pub = n.advertise<ros_erle_statusled::statusled>("statusled", 1000);
  ros_erle_statusled::statusled msg;

  msg.blue = false;
  msg.red = true;
  msg.green = true;

  ros::Rate loop_rate(10);

  int count = 0;
  while (ros::ok()){

    led_pub.publish(msg);
    ros::spinOnce();
    loop_rate.sleep();
  }  
  return 0;
}
